frame/transaction-payment/src/lib.rs

```diff
@@ -217,7 +227,11 @@ where
 
 		if positive {
 			let excess = first_term.saturating_add(second_term).saturating_mul(previous);
-			previous.saturating_add(excess).max(min_multiplier)
+			let next_multiplier = previous.saturating_add(excess).max(min_multiplier);
+			if next_multiplier > X::get() {
+				return X::get()
+			}
+			next_multiplier
```

Comment:
```suggestion
			previous
			    .saturating_add(excess)
			    .max(min_multiplier)
			    .min(max_multiplier)
```

I would just chain `.min()` on this. `return X::get()` isn't what you want.